test/Microsoft.ML.CodeAnalyzer.Tests/Code/BestFriendOnPublicDeclarationTest.cs

```diff
@@ -19,6 +19,7 @@ public sealed class BestFriendOnPublicDeclarationTest
         private readonly Lazy<string> _sourceDeclaration = TestUtils.LazySource("BestFriendOnPublicDeclaration.cs");
 
         [Fact]
+        [Obsolete]
         public async Task BestFriendOnPublicDeclaration()
```

Comment:
why we are obsoleting things in the tests? this is first time I see that :-)

Reply:
We are going to be removing most, if not all, of our internal analyzers for ML.NET. They can pretty much all be handled by the formatting tools built into runtime now.

The problem is that the version update made a bunch of our analyzers use deprecated code, but because we are going to be removing them it's not worth the time to go update all the code itself, so I just added the obsolete tag to fix the error. Is there a better way to do that? I just figured removing the analyzers in this same PR would make this PR blow up in size and so I was trying to split things off.

Reply:
Can't we suppress the analyzer warnings instead of adding obsolete?